src/tree/LocalRootTreeItemBase.ts

```diff
@@ -293,8 +294,12 @@ export abstract class LocalRootTreeItemBase<TItem extends ILocalItem, TProperty
     }
 
     private async getSortedItems(): Promise<TItem[]> {
-        const items: TItem[] = await this.getItems() || [];
-        return items.sort((a, b) => a.treeId.localeCompare(b.treeId));
+        if (ext.dockerodeError === undefined) {
```

Comment:
Why have this attached to the `extensionVariables` rather than just bubbling the error up from `refreshDockerode`?

Reply:
Because refreshDockerode gets called in two places:
1. On extension activation
1. Every time you update a docker setting

I feel like showing an error in either of those cases is too aggressive and would get annoying. I would rather delay the error and show it in a place that's actually relevant (aka the tree).

Reply:
I just don't like that it has to be manually reset to `undefined`.  I could see a case where the error is stale or not attached to the right tree node.  What happens if you update a bunch of docker settings at once??

Reply:
This is the current experience
![Screen Shot 2019-06-17 at 3 47 46 PM](https://user-images.githubusercontent.com/11282622/59641702-53efa380-9117-11e9-9d56-8e75fdf8fbe5.png)

This is what you might get if I showed the error in refreshDockerode:
![Screen Shot 2019-06-17 at 3 49 35 PM](https://user-images.githubusercontent.com/11282622/59641794-91ecc780-9117-11e9-94e6-663e04a21dec.png)

I'm open to suggestions for refactoring the code, but I definitely don't want to change the user experience.

>  I could see a case where the error is stale or not attached to the right tree node.

Worst case the user has to hit refresh and it won't be stale any more. I don't think it can be attached to the wrong tree node. This logic is unique to `LocalRootTreeItemBase`, which is the root node for all views (containers, images, volumes, etc.)

> What happens if you update a bunch of docker settings at once??


It works? I'm not entirely sure what your concern is.

Reply:
I was wondering if you could have errors from one treeItem and have it affect another valid treeItem since it's only checking of the `ext.dockerodeError` is equal to undefined.  But I'm not familiar with the code flow for this, so if it doesn't seem like a realistic scenario, I'm totally fine with dropping it.  Just wanted to get the reason since it's not a pattern I've seen you use before.